src/rest/RequestHandler.js

```diff
@@ -47,14 +55,35 @@ class RequestHandler {
     return this.execute(this.queue.shift());
   }
 
+  get globalLimited() {
+    return this.manager.globalRemaining <= 0 && Date.now() < this.manager.globalReset;
+  }
+
+  get localLimited() {
+    return this.remaining <= 0 && Date.now() < this.reset;
+  }
+
   get limited() {
-    return Boolean(this.manager.globalTimeout) || (this.remaining <= 0 && Date.now() < this.reset);
+    return this.globalLimited || this.localLimited;
   }
 
   get _inactive() {
     return this.queue.length === 0 && !this.limited && this.busy !== true;
   }
 
+  globalDelayFor(ms) {
+    return new Promise(resolve => {
+      setTimeout(
```

Comment:
```suggestion
      this.manager.client.setTimeout(
```
Attach the timeout to the client.